linetools/isgm/abssystem.py

```diff
@@ -172,10 +172,30 @@ def chk_component(self, component):
         """Additional checks on the component"""
         return True
 
+    def list_of_abslines(self):
+        """ Generate a list of the absorption lines in this system
+
+        Drawn from the components
+
+        Returns
+        -------
+        abslist : list
+
+        """
+        abslist = []
+        for component in self._components:
+            for iline in component._abslines:
+                abslist.append(iline)
+        # Return
+        return abslist
+
+
 class GenericAbsSystem(AbsSystem):
     """Class for Generic Absorption Line System
     """
     def __init__(self, radec, zabs, vlim, **kwargs):
+        if vlim is None:
+            vlim = [-500.,500.]*u.km/u.s
```

Comment:
I'm still thinking whether this is a good idea or not. If you do something like this for vlim, then for the sake of symmetry I would also allow radec and zabs to have default values, which may be useful in some cases where you do not care about those quantities. On the other hand, for the cases you do care, having default values may cause some trouble...